sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -32,65 +31,27 @@ export async function sendRequest(
   method: HttpMethods,
   url: string,
   pipeline: Pipeline,
-  options: RequestParameters = {},
+  options: InternalRequestParameters = {},
   customHttpClient?: HttpClient
 ): Promise<HttpResponse> {
   const httpClient = customHttpClient ?? getCachedDefaultHttpsClient();
   const request = buildPipelineRequest(method, url, options);
-
   const response = await pipeline.sendRequest(httpClient, request);
-
-  const rawHeaders: RawHttpHeaders = response.headers.toJSON();
-
-  const parsedBody: RequestBodyType | undefined = getResponseBody(response);
+  const headers = response.headers.toJSON();
+  const parsedBody = await getResponseBody(response, options.responseAsStream);
 
   if (options?.onResponse) {
-    options.onResponse({ ...response, request, rawHeaders, parsedBody });
+    options.onResponse({ ...response, request, rawHeaders: headers, parsedBody });
```

Comment:
One side effect of this change is we're going to start calling `onResponse` for stream responses, which we didn't do previously. I wonder if this would cause some unexpected behavior

Reply:
You're right, the intent of this property became clearer in my discussion with Jeremy despite the type obfuscation. I addressed this in https://github.com/Azure/azure-sdk-for-js/pull/27000/commits/3e49a31c2abcb74c8135c5e5f1b68572a8a132bd by making sure `parsedBody` can be either a `Record<string, uknown` or a `string` only and updated the type accordingly.